### DISCUSSION TITLE
Not work with boards with ATmega32 and ATmega1284 processors

### DISCUSSION THREAD

**Discussion by rommil67** [tttapa/Control-Surface#757] [General]
For me, the library does not work with boards with ATmega32 and ATmega1284 processors. Tiles from https://github.com/MCUdude/MightyCore#boards-manager-installation

Errors: C: \ Users \ rommi \ Documents \ Arduino \ libraries \ Control-Surface-master \ src / Submodules / Encoder / utility / interrupt_pins.h: 345: 2: error: #error "Encoder requires interrupt pins, but this board does not have any :( "
 #error "Encoder requires interrupt pins, but this board does not have any :("
  ^ ~~~~
C: \ Users \ rommi \ Documents \ Arduino \ libraries \ Control-Surface-master \ src / Submodules / Encoder / utility / interrupt_pins.h: 346: 2: error: #error "You could try defining ENCODER_DO_NOT_USE_INTERRUPTS as a kludge."
 #error "You could try defining ENCODER_DO_NOT_USE_INTERRUPTS as a kludge."
  ^ ~~~~
Use of the library Control-Surface-master version 2.0.0 from the folder: C: \ Users \ rommi \ Documents \ Arduino \ libraries \ Control-Surface-master
Using the 1.0 SPI library from the folder: C: \ Users \ rommi \ Documents \ ArduinoData \ packages \ MightyCore \ hardware \ avr \ 2.1.3 \ libraries \ SPI
exit status 1
Compilation error for the ATmega32 board.

**Comment by tttapa** [comment]
Thanks for the report. Please try using the latest `main` version, this _should_ compile without errors, even without specifying the correct number of interrupts.

**Reply by tttapa** [reply]
It looks like MCUdude's cores define a `EXTERNAL_NUM_INTERRUPTS` macro. Control Surface now recognizes this: 4b6011518fe056b4bbc289628af9be47dd5cfe35

I've just tested compilation for the ATmega32 and ATmega1284 using the latest versions of Control Surface and MightyCore, and everything seems to compile without errors or warnings.

**Reply by rommil67** [reply]
I still do not compile:

Errors:

avr-gcc: error: unrecognized command line option '-assembler-with-cpp'
exit status 1
Compilation error for the ATmega32 board.

More:
Using a pre-compiled file: C: \ Users \ rommi \ AppData \ Local \ Temp \ arduino_build_844157 \ libraries \ Control-Surface-main \ Submodules \ Encoder \ AHEncoder.cpp.o
Compiling library "SPI"
Using a pre-compiled file: C: \ Users \ rommi \ AppData \ Local \ Temp \ arduino_build_844157 \ libraries \ SPI \ SPI.cpp.o
Compiling core ...
"C: \\ Users \\ rommi \\ AppData \\ Local \\ Arduino15 \\ packages \\ arduino \\ tools \\ avr-gcc \\ 7.3.0-atmel3.6.1-arduino7 / bin / avr-gcc" -c -g -assembler-with-cpp -mmcu = atmega32 -DF_CPU = 16000000L -DARDUINO = 10815 -DARDUINO_AVR_ATmega32 -DARDUINO_ARCH_AVR "-IC: \\ Users \\ rommi \\ AppData \\ Local \\ \ Arduino15 \\ packages MightyCore \\ hardware \\ avr \\ 2.1.3 \\ cores \\ MCUdude_corefiles "" -IC: \\ Users \\ rommi \\ AppData \\ Local \\ Arduino15 \\ packages \\ MightyCore \\ hardware \\ avr \\ 2.1.3 \\ variants \\ standard "" C: \\ Users \\ rommi \\ AppData \\ Local \\ Arduino15 \\ packages \\ MightyCore \\ hardware \\ avr \\ 2.1.3 \\ cores \\ MCUdude_corefiles \\ wiring_pulse.S "-o" C: \\ Users \\ rommi \\ AppData \\ Local \\ Temp \\ arduino_build_844157 \\ core \\ wiring_pulse.S.o "
avr-gcc: error: unrecognized command line option '-assembler-with-cpp'
Using the library Control-Surface-main version 2.0.0 from the folder: C: \ Users \ rommi \ OneDrive \ Documents \ Arduino \ libraries \ Control-Surface-main
Using the 1.0 SPI library from the folder: C: \ Users \ rommi \ AppData \ Local \ Arduino15 \ packages \ MightyCore \ hardware \ avr \ 2.1.3 \ libraries \ SPI
exit status 1
Compilation error for the ATmega32 board.

**Reply by rommil67** [reply]
It's ok! I installed the Arduino IDE on a fresh win10 system and it's OK. Thanks Peter
